lib/modules/manager/npm/post-update/npm.ts

```diff
@@ -123,6 +124,26 @@ export async function generateLockFile(
 
     // Read the result
     lockFile = await readFile(upath.join(cwd, filename), 'utf8');
+
+    // Massage lockfile counterparts of package.json that were modified
+    // because npm install was called with an explicit version for rangeStrategy = update-lockfile
+    if (lockUpdates.length) {
+      let detectedIndent: string;
+      let lockFileParsed: any;
+      try {
+        detectedIndent = detectIndent(lockFile).indent || '  ';
+        lockFileParsed = JSON.parse(lockFile);
+      } catch (err) {
+        logger.warn({ err }, 'Error parsing npm lock file');
+      }
+      if (lockFileParsed && lockFileParsed?.lockfileVersion === 2) {
+        lockUpdates.forEach((lockUpdate) => {
+          lockFileParsed.packages[''][lockUpdate.depType][lockUpdate.depName] =
```

Comment:
We need to add a `if (lockFileParsed.packages?.['']?.[lockUpdate.depType]?.[lockUpdate.depName])` defensive check first